### ISSUE TITLE
tf.contrib.image.connected_components works incorrectly with 3D numpy arrays

### ISSUE THREAD

**Issue by dpaniukov** [tensorflow/tensorflow#18917]
### System information
Have I written custom code: N/A
OS Platform and Distribution: Linux Ubuntu 16.04
Bazel version: N/A
TensorFlow installed from binary
TensorFlow version: v1.7.0-3-g024aecf414 1.7.0
Python version: 3.6
CUDA/cuDNN version: CUDA 9.0 / cuDNN 7.1.1
GPU model and memory: GeForce GTX 1060 6GB

### Exact command to reproduce:
`import numpy as np`
`from tensorflow.contrib.image import connected_components`
`a = np.zeros((2,3,4))`
`a[0,0,0]=1`
`a[0,0,1]=1`
`a[1,0,1]=1`
`print((tf.Session().run(connected_components(a))))`

> [[[1 1 0 0]
  [0 0 0 0]
  [0 0 0 0]]
 [[0 2 0 0]
  [0 0 0 0]
  [0 0 0 0]]]

It gives two components instead of one.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
Bazel version
CUDA/cuDNN version
Exact command to reproduce

**Comment by dpaniukov** [comment]
Updated, thanks.

**Comment by dpaniukov** [comment]
@ntenenz Thanks for the clarification! I work with 3D images of brains and need this to work with a single 3D image. At some point of the analysis, I search for clusters of connected voxels (3D pixels). As of now, to accomplish the task I use:

`from scipy.sparse.csgraph import connected_components`
`from sklearn.neighbors import radius_neighbors_graph`

`n_components, labels = connected_components(radius_neighbors_graph(my_voxels, radius=1, mode='connectivity', include_self=True).toarray())`

Will it be possible to incorporate such or similar functionality in tensorflow? It will significantly help researchers like me and speed things up with a GPU. 

Thanks for your consideration.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 77 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 92 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 107 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by bignamehyp** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
